docs/notebooks/batch_optimization.pct.py

```diff
@@ -0,0 +1,176 @@
+
+    return {
+        OBJECTIVE: {
+            "model": gpr,
+            "optimizer": gpflow.optimizers.Scipy(),
+            "optimizer_args": {
+                "minimize_args": {"options": dict(maxiter=100)},
+            },
+        }
+    }
+
+
+model_spec = build_model(initial_data[OBJECTIVE])
+models = map_values(create_model, model_spec)
+
+# %% [markdown]
+# ## Batch acquisition functions.
+# To peform batch BO, we must define a batch acquisition function. Two popular batch acquisition functions supported in Trieste are `BatchMonteCarloExpectedImprovement` and `LocallyPenalizedExpectedImprovement`. Although both of these acquisition functions recommend batches of diverse query points, the batches are chosen in very different ways. `BatchMonteCarloExpectedImprovement` jointly allocates the batch of points as those with the largest expected improvement over our current best solution. In contrast, `LocallyPenalizedExpectedImprovement` greedily builds the batch, sequentially adding the maximizers of the standard (non-batch) `ExpectedImprovement` function penalized around the current pending batch points. In practice, `BatchMonteCarloExpectedImprovement` has superior performance for small batches (`batch_size`<10) but scales poorly for larger batches.
+#
+# Note that both of these acquisition functions have controllable parameters. In particular, `BatchMonteCarloExpectedImprovement` is computed using a Monte-Carlo method (so it requires a `sample_size`), but uses a reparametrisation trick to make it deterministic. `LocallyPenalizedExpectedImprovement` has parameters controlling the degree of penalization that must be estimated from a random sample of  `num_samples` model predictions.
+#
+# We now visualize the batch of 10 points chosen by each of these methods overlayed on the standard `ExpectedImprovement` acquisition function. `BatchMonteCarloExpectedImprovement` chooses a more diverse set of points, whereas `LocallyPenalizedExpectedImprovement` focuses evaluations into more promising areas of the space.
+
+# %%
+from trieste.acquisition import (
+    BatchMonteCarloExpectedImprovement,
+    ExpectedImprovement,
+    LocallyPenalizedExpectedImprovement,
+)
+from trieste.acquisition.rule import EfficientGlobalOptimization
+
+ei = ExpectedImprovement()
+ei_acq_function = ei.using(OBJECTIVE).prepare_acquisition_function(initial_data, models)
+plot_acq_function_2d(ei_acq_function, [0, 0], [1, 1], contour=True, grid_density=100)
+
+batch_ei = BatchMonteCarloExpectedImprovement(sample_size=1000)
+chosen_points, _ = EfficientGlobalOptimization(
+    num_query_points=10, builder=batch_ei.using(OBJECTIVE)
+).acquire(search_space, initial_data, models)
+plt.scatter(
+    chosen_points[:, 0],
+    chosen_points[:, 1],
+    color="red",
+    lw=5,
+    label="Batch-EI",
+    marker="*",
+    zorder=1,
+)
+
+local_penalization = LocallyPenalizedExpectedImprovement(search_space, num_samples=1000)
+chosen_points, _ = EfficientGlobalOptimization(
```

Comment:
same here, it can actually be surprising that we use the same rule for a greedy approach